05.wc/wc.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = parse_options
+  if ARGV.empty?
+    handle_stdin(options)
+  else
+    handle_files(ARGV, options)
+  end
+end
+
+def parse_options
+  options = { lines: false, words: false, bytes: false }
+  OptionParser.new do |opts|
+    opts.banner = 'Usage: wc.rb [options] [file ...]'
+    opts.on('-l', 'Count lines') { options[:lines] = true }
+    opts.on('-w', 'Count words') { options[:words] = true }
+    opts.on('-c', 'Count bytes') { options[:bytes] = true }
+  end.parse!
+  options.values.any? ? options : options.transform_values { true }
+end
+
+def handle_stdin(options)
+  file_content = $stdin.read
+  output_counts_hash = { counts: count_all(file_content) }
+  output_results(output_counts_hash, options)
+end
+
+def handle_files(files, options)
+  file_counts = files.map { |file| count_file_contents(file) }
```

Comment:
```suggestion
  file_counts = files.map do |file|
    file_content = File.read(file)
    { counts: count_all(file_content), filename: file }
  end
```

DRYに貢献するわけではないのでメソッド化しなくてもいいかも（好みの問題）

Reply:
@JunichiIto 
かしこまりました🙇‍♂️